src/libraries/System.Console/src/System/ConsolePal.Windows.cs

```diff
@@ -676,6 +676,25 @@ public static unsafe string Title
         }
 
         public static void Beep()
+        {
+            const char BellCharacter = '\u0007'; // Windows doesn't use terminfo, so the codepoint is hardcoded.
+
+            if (!Console.IsOutputRedirected)
+            {
+                Console.Out.Write(BellCharacter);
```

Comment:
If we _really_ want to do what this PR is doing, we shouldn't be going to Console.Out. We should instead do what the unix implementation does, which is going directly to the stdout handle / file descriptor, such that it won't target Console.Out and any custom writer it may have been redirected to, e.g.
https://github.com/dotnet/runtime/blob/25a50853236dbec65408c7942793df3eb2e8cdbc/src/libraries/System.Console/src/System/ConsolePal.Unix.cs#L1121
In the meantime, though, I suggest this PR be reverted.

Reply:
> In the meantime, though, I suggest this PR be reverted.

I am going to give it a try and if I can not implement it in 1h I am going to revert the PR. Does that sound OK @stephentoub ?

Reply:
Yup, thanks.

Reply:
That said, I'm still not sure about the premise of this change. Isn't it strange that Beep with and without parameters are now fundamentally different? This goes to Jan's points about this being a breaking change.

Reply:
> That said, I'm still not sure about the premise of this change. 

https://github.com/dotnet/runtime/issues/102211#issuecomment-2110973504

Reply:
> > That said, I'm still not sure about the premise of this change.
> 
> [#102211 (comment)](https://github.com/dotnet/runtime/issues/102211#issuecomment-2110973504)

I know what's written in the issue. What I meant was I'm not sure it's a desirable change.

Reply:
I've sent https://github.com/dotnet/runtime/pull/104966